services/question/README.md

```diff
@@ -0,0 +1,428 @@
+# Question Service User Guide
+
+## Pre-requisites
+
+You need to create the `.env` files from the provided `.env.sample` files in both the root directory and
+the `services/question` directory.
+
+1. Open the Command Prompt.
+2. Navigate to the root directory of the project.
+3. Run the following command to create the `.env` files:
+```cmd
+cp .env.sample .env
+cp services/question/.env.sample services/question/.env
+```
+4. After setting up the .env files, build the Docker images and start the containers using the following command:
+```cmd
+docker compose build
+docker compose up -d
+```
+5. To stop and remove the containers and associated volumes, use the following command:
+```cmd
+docker compose down -v
+```
+
+## Get Questions
+
+This endpoint allows the retrieval of all the questions in the database. If filter by (optional) parameters, questions
+that matches with parameters will be returned; if no parameters are provided, all questions will be returned.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions`
+
+### Parameters:
+
+- `title` (Optional) - Filter by question title.
+- `description` (Optional) - Filter by question description.
+- `topics` (Optional) - Filter by topics associated with the questions.
+- `difficulty` (Optional) - Filter by question difficulty.
+
+### Responses:
+
+| Response Code               | Explanation                                |
+|-----------------------------|--------------------------------------------|
+| 200 (OK)                    | Success, all questions are returned        |
+| 404 (Not Found)             | No questions found.                        |
```

Comment:
If no questions are found, I think it makes more sense to return a response code of 200 with an empty array.

Reply:
Ahh, but then how about the other endpoints? If lets say the question is not found, is it better to return an empty array or return an error message? Because currently right now for other endpoints, if its not found, an error message is returned instead.

Reply:
I think it's fine to 404 for get question by id and delete question (since they are providing an id). For random question not too sure, I guess either way works?

Reply:
Okay, I think I will return an empty JSON for getQuestions and getQuestionsByParameters (if not found)